src/ansys/sherlock/core/model.py

```diff
@@ -632,3 +644,265 @@ def createExportTraceCopperLayerParams(
+                if not isinstance(max_edge_length, MaxEdgeLength):
+                    raise SherlockExportFEAModelError(message="Maximum edge length is invalid.")
+
+            if detect_lead_modeling not in ["ENABLED", "enabled", "DISABLED", "disabled"]:
+                raise SherlockExportFEAModelError(message="Detect lead modeling status is invalid.")
+
+            for param in lead_model_parameters:
+                lead_modeling = param.get("lead_modeling")
+                if lead_modeling not in ["ENABLED", "enabled", "DISABLED", "disabled"]:
+                    raise SherlockExportFEAModelError(message="Lead modeling status is invalid.")
+
+                lead_element_order = param.get("lead_element_order")
+                if lead_element_order not in [
+                    "First Order (Linear)",
+                    "Second Order (Quadratic)",
+                    "Solid Shell",
+                ]:
+                    raise SherlockExportFEAModelError(message="Lead element order is invalid.")
+
+                max_mesh_size = param.get("max_mesh_size")
+                if not isinstance(max_mesh_size, MaxMeshSize):
+                    raise SherlockExportFEAModelError(message="Maximum mesh size is invalid.")
+
+                vertical_mesh_size = param.get("vertical_mesh_size")
+                if not isinstance(vertical_mesh_size, VerticalMeshSize):
+                    raise SherlockExportFEAModelError(message="Vertical mesh size is invalid.")
+
+                thickness_count = param.get("thicknessCount", 3)
+                if (
+                    not isinstance(thickness_count, int)
+                    or thickness_count < 1
+                    or thickness_count > 5
+                ):
+                    raise SherlockExportFEAModelError(
+                        message="Invalid thickness count. Must be an integer between 1 and 5, with "
+                        "a default value of 3."
+                    )
+
+                aspect_ratio = param.get("aspectRatio", 2)
+                if not isinstance(aspect_ratio, int) or aspect_ratio < 1 or aspect_ratio > 10:
+                    raise SherlockExportFEAModelError(
+                        message="Invalid aspect ratio. Must be an integer between 1 and 10, with "
+                        "a default value of 2."
+                    )
+
+            if not isinstance(display_model, bool):
+                raise SherlockExportFEAModelError(message="display_model must be a boolean.")
+
+            if not isinstance(clear_FEA_database, bool):
+                raise SherlockExportFEAModelError(message="clear_FEA_database must be a boolean.")
+
+            if not isinstance(use_FEA_model_id, bool):
+                raise SherlockExportFEAModelError(message="use_FEA_model_id must be a boolean.")
+
+            if not isinstance(coordinate_units, str):
+                raise SherlockExportFEAModelError(message="coordinate_units must be a string.")

```

Comment:
I don't think you need to validate types for parameters that are only used in the request since the GRPC code will cause an error if the wrong type is used in the request. If the parameter needs to be a particular type for the PySherlock code, then it can help to have type validation since the error may not be as obvious.